pkg/symbolizer/symbolizer.go

```diff
@@ -55,22 +69,57 @@ type DebugInfoFetcher interface {
 
 func New(
 	logger log.Logger,
+	reg *prometheus.Registry,
 	metastore pb.MetastoreServiceClient,
 	debuginfo DebugInfoFetcher,
 	symbolizer *symbol.Symbolizer,
 	debuginfodCacheDir string,
 	debuginfoCacheDir string,
 	batchSize uint32,
 ) *Symbolizer {
-	return &Symbolizer{
+	attemptsTotal := prometheus.NewCounter(
+		prometheus.CounterOpts{
+			Name: "symbolizer_attempts_total",
+			Help: "Total number of symbolization attempts in batches.",
+		},
+	)
+	errorsTotal := prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Name: "symbolizer_errors_total",
+			Help: "Total number of symbolization errors in batches, partitioned by an error reason.",
+		},
+		[]string{"reason"},
+	)
+	duration := prometheus.NewHistogram(
+		prometheus.HistogramOpts{
+			Name:    "symbolizer_duration_seconds",
+			Help:    "How long it took in seconds to finish a round of the symbolization cycle in batches.",
+			Buckets: []float64{0.001, 0.01, 0.1, 0.3, 0.6, 1, 3, 6, 9, 20, 30, 60, 90, 120},
+		},
+	)
+	storeDuration := prometheus.NewHistogram(
+		prometheus.HistogramOpts{
+			Name:    "symbolizer_store_duration_seconds",
+			Help:    "How long it took in seconds to store a batch of the symbolized locations.",
+			Buckets: []float64{0.001, 0.01, 0.1, 0.3, 0.6, 1, 3, 6, 9, 20, 30, 60, 90, 120},
+		},
+	)
+	reg.MustRegister(attemptsTotal, errorsTotal, duration, storeDuration)
```

Comment:
I would rather suggest using `promauto` package and registering in-line to be consistent with the rest of the code base.

Reply:
Good point, I have found a few examples in parca-agent repository.